pkg/controllers/provisioning/scheduling/nodeclaim.go

```diff
@@ -254,6 +189,22 @@ func filterInstanceTypesByRequirements(instanceTypes []*cloudprovider.InstanceTy
 		// any errors.
 		if itCompat && itFits && itHasOffering {
 			results.remaining = append(results.remaining, it)
+		} else {
+			if !itFits && !itHasOffering && !results.requirementsMet {
+				results.reason = fmt.Sprintf("%s, %s, %s, %s", it.Name, itUnfitReason, offeringReason, it.Requirements.Intersects(requirements))
```

Comment:
I'm not sure about this.  The "doesn't fit" reason can vary across the instance types, and this will log only the last one.

Reply:
it is looping across all the instance types right

Reply:
And overwriting the "reason" with each one, so only the last one will be returned.   Not sure we want to display all reasons as that would be really long, but displaying only the last doesn't seem correct either.

Reply:
seems logical yeah , how about writing to some log file ?

Reply:
I changed it to display all messages considering . The message we are writing is short and would probably take 1-2 line and this way the user might be fully aware about his resources too